### DISCUSSION TITLE
Update Function

### DISCUSSION THREAD

**Discussion by PergonGER** [Lailloken/Exile-UI#171] [Ideas & Feedback]
Hey,
it's me again.
Would be great if you could implement a update function to your tool.
Showing a changelog after update internally or in browser.

Thanks!

**Comment by Lailloken** [comment]
Thanks for your suggestion.

I have intentionally avoided implementing an updater because, imo, the potential risks outweigh the benefits:
- this league-launch and the increased traffic has shown that many people have problems with insufficient write-permissions, which would make an auto-update fail every time (and also increase bug reports)
- I sometimes make drastic changes, so reading the notes is often important. An updater increases the probability of people blindly clicking 'update' without reading them (I often do it myself in PoB, for example), then not knowing what changed and potentially submitting bug reports
- some new additions require screenshots or a short video-clip to show how something works or needs to be configured, and there's no way to embed videos into an updater

That's why I chose the current method of just showing a small notification that redirects to the releases page. It's way simpler for me, I make sure the user at least scrolls past the changelogs, I can include media, and it's not too much work for the user. I may include a short summary in the future that will also show in the notification, stating what the update includes (e.g. "important hotfix" or "new feature", etc.).

**Comment by PergonGER** [comment]
Thanks for your reply.
Guess you are right, i am just lazy :D

Will do it with the github updater then :-P Why didnt i think earlier of
this?

Have a nice day!

Am Mo., 19. Dez. 2022 um 10:02 Uhr schrieb Lailloken <
***@***.***>:
